staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
Nice. Note, though, this annoying detail: we're really looking for the channel (or the `Context`'s channel) to be _available_, not closed, though the latter implies the former. That is, whether or not the channel is closed, if there's an item available to be received, the function should still stop. We don't expect the function operate like this, demanding actual closure:
```go
for {
  if _, ok := <-ch; !ok {
    return
  }
}
```
Instead, we expect just this:
```go
for {
  select {
  // Other things ...
  case <-ch:
    return
  }
}
```
Hence, consider writing "is available" rather than "is closed" in the documentation.